modules/minimint-wallet/src/bitcoind.rs

```diff
@@ -99,8 +111,7 @@ pub mod test {
     use bitcoin::hashes::Hash;
     use bitcoin::{Block, BlockHash, BlockHeader, Network, Transaction};
     use std::collections::{HashMap, VecDeque};
-    use std::sync::Arc;
-    use tokio::sync::Mutex;
+    use std::sync::{Arc, Mutex};
```

Comment:
> Did you check they aren't held across awaits and how do we prevent people from doing so?

clippy has a check for that, but also checked manually now. We are only holding mutex internal, i.e. never return or await with mutex held. 

Regarding blocking, it can be a performance issue but not a correctness issue. In this case, it is only for testing and we don't block for too long.

Reply:
Ok, perfect, then let's go with this until it becomes a problem (likely never).